lib/webauthn/authenticator_response.rb

```diff
@@ -82,11 +82,26 @@ def valid_challenge?(expected_challenge)
       OpenSSL.secure_compare(client_data.challenge, expected_challenge)
     end
 
+    # @return [Boolean]
+    # @param [String, Array<String>] expected_origin
+    # Validate if origin configured for RP is matching the one received from client
     def valid_origin?(expected_origin)
-      expected_origin && (client_data.origin == expected_origin)
+      return false unless expected_origin
+
+      case expected_origin
+      when Array
+        expected_origin.include?(client_data.origin) # allow multiple origins as per spec
+      else
+        client_data.origin == expected_origin # keep the default behaviour for backwards compatibility
+      end
```

Comment:
It's kinda confusing IMO for the `origin` attribute to be either a string or a hash – the name kinda implies that it should be just one.

I was thinking that an option could be to accept the `origin` attribute as we historically have but add a new `allowed_origins` config that will take precedence over the `origin` attribute. Then in a future major version we could deprecate the `origin` attribute and leave only the `allowed_origins`.

What do you think?

Reply:
hi @santiagorodriguez96 , that is a good question. I was also thinking about having `allowed_origins` config but then rejected this idea because it feels redundant to introduce multiple way to assign a value to a single control that we want to have. Also it would probably cause a mental break for people reading documentation trying to understand how `origin` is different from `allowed_origins`. Though, if the intention is to deprecate `origin` config in future versions - we can of course stick to `allowed_origins` long term. I will work on this change in scope of this PR